core/src/io/anuke/mindustry/world/meta/Bottleneck.java

```diff
@@ -0,0 +1,21 @@
+package io.anuke.mindustry.world.meta;
+
+import io.anuke.arc.graphics.*;
+
+/**
+ * "port" of https://mods.factorio.com/mods/trold/Bottleneck
+ */
+public enum Bottleneck{
+    none,   // green, doing fine <3
+    output, // yellow, limited by output
+    input;  // red, limited by input
+
+    public static Color color(Bottleneck state){
```

Comment:
`color` should be a field of the `Bottleneck` class and passed into its constructor. 
```java
public final Color color;

Bottleneck(Color color){
    this.color = color;
}
```

Reply:
Ehmm, i'm not entirely sure how to construct an enum that way, is there another class in this repo that i could study for that?